snownews: add openssl dependency

apjanke opened the pull request:
Fixes build failure on 10.11.

DomT4 commented:
@BrewTestBot test this please

DomT4 commented:
Didn't link correctly on El Cap:

```
        Error: 1 problem in 1 formula
snownews:
 * object files were linked against system openssl
These object files were linked against the deprecated system OpenSSL.
Adding `depends_on "openssl"` to the formula may help.
  /usr/local/Cellar/snownews/1.5.12_1/bin/snownews
```

Probably needs one of those `--with-ssl=` type arguments.

apjanke commented:
My bad: I had been running `brew audit` after editing the formulae but before installing for these. I can reproduce the audit failure on my machine. Looking in to it.

apjanke commented:
Fixed; `brew audit` is clean on my machine now.

There's no `--with-ssl=`, since it's using a hand-rolled `./configure` script. You have to pass it openssl with a `-L` in the `EXTRA_LDFLAGS` configure variable. I had to patch the `configure` script to move `$EXTRA_LDFLAGS` up ahead of the `-l` flags so the formula-specified path would take precedence over the default system path for `-lcrypto`.

I also removed the old patch. Its comments were talking about fixing a link error on OS X, but the section actually patched was the Linux section in `./configure`. I'm guessing that's vestigial or a bug; let me know if it should actually be kept in.

dunn commented on Library/Formula/snownews.rb:
```diff
@@ -3,16 +3,16 @@ class Snownews < Formula
-  # homebrew reference added on 2013-07-06 to https://kiza.eu/software/snownews/downloading
+  # Fix system openssl linking error on OS X
+  # Allow EXTRA_LDFLAGS to take precedence so we can link brewed openssl
+  # instead of deprecated system openssl
```
Could you report this upstream? Thanks!

apjanke replied:
I reported it upstream by emailing the author, per [its web page](https://kiza.eu/software/snownews/faq#Feedback_and_bug_reporting).

apjanke commented:
Well, that's a weird failure. On 10.9 and 10.10, it's not linking to `zlib`. Not even attempting to.

The failed 10.9 test-bot build:

```
clang main.o netio.o interface.o xmlparse.o updatecheck.o conversions.o dialog.o ui-support.o categories.o about.o cookies.o setup.o net-support.o digcalc.o filters.o io-internal.o os-support.o zlib_interface.o support.o -o snownews -L/usr/local/opt/openssl/lib -lncurses -lcrypto -lxml2  -lintl  -liconv
[ ... snip ... ]
Undefined symbols for architecture x86_64:
  "_inflate", referenced from:
      _jg_zlib_uncompress in zlib_interface.o
  "_inflateEnd", referenced from:
      _jg_zlib_uncompress in zlib_interface.o
  "_inflateInit2_", referenced from:
      _jg_zlib_uncompress in zlib_interface.o
ld: symbol(s) not found for architecture x86_64
```

The link step on my 10.11 box:
```
clang main.o netio.o interface.o xmlparse.o updatecheck.o conversions.o dialog.o ui-support.o categories.o about.o cookies.o setup.o net-support.o digcalc.o filters.o io-internal.o os-support.o zlib_interface.o support.o -o snownews -L/usr/local/opt/openssl/lib -lncurses -lcrypto -L/Applications/Xcode-7-GM.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/lib -lxml2 -lz -lpthread -licucore -lm  -lintl  -liconv
```

Looks like that old zlib fix that I thought was vestigial and removed here is still needed on earlier OS versions, and I didn't realize it because I tested it on a 10.11 box. I can reproduce on my 10.9 box. I'll figure out what needs to be added back in.

apjanke commented:
I just stuck the old zlib workaround functionality back in. (I had misread the patch file earlier about it being for Linux; it's actually in the BSD section.) I made it unconditional. This means there will be a redundant `-lz` in the link command on 10.11, which is no big deal.

I think I know why it's happening and why 10.11 is different. An `-lz` is always required when building snownews. But it's not directly supplied by `configure`. On OS X 10.11, and I assume other systems, it's coming from `xml2-config --libs`.

```
$ xml2-config --libs
-L/Applications/Xcode-7-GM.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/lib -lxml2 -lz -lpthread -licucore -lm
```

On OS X 10.9 and 10.10, `xml2-config --libs` does not provide it.
```
$ xml2-config --libs
-lxml2
```

I changed the formula to pass `-lz` using `$EXTRA_LDFLAGS` instead of through a patch; I think it's more readable that way, and we seem to prefer minimizing the use of patches.

This now builds on both 10.9 and 10.11 for me.

DomT4 commented:
Merged in https://github.com/Homebrew/homebrew/commit/d274d6cfc61f0a8fc51231fc541b8918a8d9f16c. Thanks @apjanke!
